aya/src/bpf.rs

```diff
@@ -615,6 +662,7 @@ impl<'a> BpfLoader<'a> {
         Ok(Bpf {
             maps: maps?,
             programs,
+            _btf_fd: btf_fd,
```

Comment:
This was previously being leaked as we would create it but no one would own it/close it. My gut feeling is that `ProgramData` and `MapData` should stop storing the raw fd and instead *use* a a borrowedfd when loading and then drop it. This does change the API slightly as re-loading maps/programs would require the btf fd being passed though.